File: com.mini.util/src/main/java/com/mini/util/digest/MD5Util.java
package com.mini.util.digest;

import java.io.InputStream;

public class MD5Util {
    private static final DigestUtil instance = new DigestUtil("MD5");

    /**
     * MD5加密
     *
     * @param data
     * @return
     */
    public static byte[] encodeToByte(String data) {
        return instance.encodeToByte(data);
    }

    /**
     * MD5加密
     *
     * @param data
     * @return
     */
    public static String encode(String data) {
        return instance.encode(data);
    }

    /**
     * MD5加密
     *
     * @param data
     * @return
     */
    public static byte[] encodeToByte(byte[] data) {
        return instance.encodeToByte(data);
    }

    /**
     * MD5加密
     *
     * @param data
     * @return
     */
    public static String encode(byte[] data) {
        return instance.encode(data);
    }

    /**
     * MD5加密
     *
     * @param data
     * @return
     */
    public static byte[] encodeToByte(InputStream data) {
        return instance.encodeToByte(data);
    }

    /**
     * MD5加密
     *
     * @param data
     * @return
     */
    public static String encode(InputStream data) {
        return instance.encode(data);
    }

}

File: sn.mini.kotlin.util/src/main/java/sn/mini/kotlin/util/degest/MD5Util.kt
package sn.mini.kotlin.util.degest

import java.io.InputStream

object MD5Util {
    private val instance = DigestUtil("MD5")

    /**
     * MD5加密
     *
     * @param data
     * @
     */
    fun encodeToByte(data: String): ByteArray = instance.encodeToByte(data)


    /**
     * MD5加密
     *
     * @param data
     * @
     */
    fun encode(data: String): String = instance.encode(data)

    /**
     * MD5加密
     *
     * @param data
     * @
     */
    fun encodeToByte(data: ByteArray): ByteArray = instance.encodeToByte(data)

    /**
     * MD5加密
     *
     * @param data
     * @
     */
    fun encode(data: ByteArray): String = instance.encode(data)

    /**
     * MD5加密
     *
     * @param data
     * @
     */
    fun encodeToByte(data: InputStream): ByteArray = instance.encodeToByte(data)

    /**
     * MD5加密
     *
     * @param data
     * @
     */
    fun encode(data: InputStream): String = instance.encode(data)

}
